src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,22 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    Map<String, UserData> cache = new ConcurrentHashMap<>();
```

Comment:
The 'cache' field should be private to encapsulate the internal state of the class and prevent unauthorized access or modification from outside the class.